packages/datadog-plugin-apollo-gateway/src/plan.js

```diff
@@ -0,0 +1,44 @@
+
+'use strict'
+
+const TracingPlugin = require('../../dd-trace/src/plugins/tracing')
+const { storage } = require('../../datadog-core')
+
+class ApolloGatewayPlanPlugin extends TracingPlugin {
+  static get id () { return 'apollo-gateway' }
+  static get operation () { return 'plan' }
+  static get type () { return 'apollo-gateway' }
+  static get kind () { return 'server' }
+
+  static get prefix () {
+    return 'tracing:apm:apollo-gateway:plan'
+  }
+
+  bindStart (ctx) {
+    const store = storage.getStore()
+    const childOf = store ? store.span : null
+
+    const spanData = {
+      childOf,
+      service: this.config.service,
+      type: this.constructor.type,
+      kind: this.constructor.kind,
+      meta: {}
+    }
+
+    const span = this.startSpan(`${this.constructor.id}.${this.constructor.operation}`, spanData, false)
+    ctx.parentStore = store
+    ctx.currentStore = { ...store, span }
+
+    return ctx.currentStore
+  }
+  end (ctx) {
+    ctx.currentStore.span.finish()
```

Comment:
Again similar to the above all of these should have additional checks.

Reply:
I wonder also if they could be added to the base class at this point to avoid having to change this code everywhere. I could be wrong, but I don't think there is a conflict anywhere between finish and end, although error might be a bit more problematic. This is out of scope of this PR though, but if it ends up being an easy change it could be done now to simplify it since there are many plugins involved.